internal/api/finalizers.go

```diff
@@ -78,7 +78,7 @@ func PatchOwnerReferences(ctx context.Context, c client.Client, obj client.Objec
 	return nil
 }
 
-func patchAnnotation(ctx context.Context, c client.Client, obj client.Object, key, value string) error {
+func PatchAnnotation(ctx context.Context, c client.Client, obj client.Object, key, value string) error {
```

Comment:
Not at all a new problem introduced by this PR, but now might be a good time to move this function out of `finalizers.go`, which seems like an odd place for it, and into `annotations.go` here in this same package.

Reply:
Actually... do we really need to export this? See https://github.com/akuity/kargo/pull/4000/files#r2076518451

Reply:
> might be a good time to move this function...

If indeed we do not need to modify this function at all, then don't bother moving it.